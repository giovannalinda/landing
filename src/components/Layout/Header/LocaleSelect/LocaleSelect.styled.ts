import { motion } from 'framer-motion'
import { FiChevronDown } from 'react-icons/fi'
import styled, { css } from 'styled-components'
import { theme } from '~/styles'

type ChevronDownWithMarginProps = {
  isOpen: boolean
}

export const Container = styled.div`
  position: relative;

  button {
    display: flex;
    align-items: center;

    &:focus {
      outline: none;
    }

    &:hover {
      transform: none;
      opacity: 1;
    }
  }

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    margin-right: ${theme.space[4]};

    button {
      padding: 0;
      height: auto;
      background: transparent;
    }
  }
`

export const CountryFlag = styled.img`
  width: 3.2rem;
  border-radius: ${theme.borderRadius.md};
`

export const ChevronDownWithMargin = styled(
  FiChevronDown,
)<ChevronDownWithMarginProps>`
  margin-left: ${theme.space[4]};
  color: ${theme.colors.gray500};
  transition: ${theme.transitions.fast};

  ${({ isOpen }) =>
    isOpen &&
    css`
      transform: rotate(-180deg);
    `}
`

export const SelectList = styled(motion.ul)`
  position: absolute;
  padding: ${theme.space[3]};
  top: calc(100% - 0.4rem);
  background: ${theme.colors.gray800};
  border-radius: 0 0 ${theme.borderRadius.lg} ${theme.borderRadius.lg};

  li + li {
    margin-top: ${theme.space[4]};
  }

  a {
    display: flex;
    align-items: center;
    color: ${theme.colors.white};
    font-weight: ${theme.font.weigths.medium};
    font-size: ${theme.font.sizes.paragraph};
  }
`
