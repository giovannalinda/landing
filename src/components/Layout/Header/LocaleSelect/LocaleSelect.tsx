import { useState } from 'react'
import { useRouter } from 'next/router'
import Link from 'next/link'
import { AnimatePresence } from 'framer-motion'

import { Button } from '~/components'

import * as S from './LocaleSelect.styled'

const locales = [
  {
    name: 'English',
    locale: 'en-US',
    icon: '/countries/usa.svg',
  },
  {
    name: 'PortuguÃªs',
    locale: 'pt-BR',
    icon: '/countries/brazil.svg',
  },
]

export function LocaleSelect() {
  const router = useRouter()

  const [isOpen, setIsOpen] = useState(false)

  function toggleSelectListVisibility() {
    setIsOpen((prevState) => !prevState)
  }

  const localeSelected =
    locales.find((language) => language.locale === router.locale) || locales[0]

  return (
    <S.Container>
      <Button
        variant='neutral'
        size='sm'
        onClick={toggleSelectListVisibility}
        aria-label='locale select'
      >
        <S.CountryFlag
          width={70}
          height={30}
          alt={localeSelected.name}
          src={localeSelected.icon}
        />
      </Button>
      <AnimatePresence>
        {isOpen && (
          <S.SelectList
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            aria-label='select list'
          >
            {locales
              .filter((locale) => locale.name !== localeSelected.name)
              .map(({ name, icon, locale }) => (
                <li key={locale}>
                  <Link href='/' locale={locale}>
                    <S.CountryFlag src={icon} alt={name} />
                  </Link>
                </li>
              ))}
          </S.SelectList>
        )}
      </AnimatePresence>
    </S.Container>
  )
}
