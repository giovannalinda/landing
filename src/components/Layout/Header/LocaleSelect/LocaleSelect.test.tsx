import { render, screen, userEvent } from '~/__helpers__/app-tests'
import { LocaleSelect } from './LocaleSelect'

describe('LocaleSelect', () => {
  it('should match snapshot', () => {
    const { container } = render(<LocaleSelect />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should open list', async () => {
    render(<LocaleSelect />)

    userEvent.click(
      await screen.findByRole('button', {
        name: 'locale select',
      }),
    )

    expect(
      await screen.findByRole('list', {
        name: 'select list',
      }),
    ).toBeInTheDocument()
  })
})
