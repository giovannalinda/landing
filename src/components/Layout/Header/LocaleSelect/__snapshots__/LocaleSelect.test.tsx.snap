// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleSelect should match snapshot 1`] = `
<div
  class="sc-lbxAil hfLLGD"
>
  <button
    aria-label="locale select"
    class="sc-gsnTZi kcqiXn"
  >
    <img
      alt="English"
      class="sc-iNWwEs fIcmmO"
      height="30"
      src="/countries/usa.svg"
      width="70"
    />
  </button>
</div>
`;
